scripts/ci/stats/merged_prs.py

```diff
@@ -0,0 +1,117 @@
+        formatter_class=argparse.RawDescriptionHelpFormatter, allow_abbrev=False)
+
+    parser.add_argument('--pull-request', required=True, help='pull request number', type=int)
+    parser.add_argument('--repo', required=True, help='github repo')
+
+    return parser.parse_args()
+
+def gendata(data, index):
+    for t in data:
+        yield {
+                "_index": index,
+                "_source": t
+                }
+
+def main():
+    args = parse_args()
+    token = os.environ.get('GITHUB_TOKEN')
+    if not token:
+        sys.exit('Github token not set in environment, please set the '
+                 'GITHUB_TOKEN environment variable and retry.')
+
+    gh = Github(token)
+    json_list = []
+    gh_repo = gh.get_repo(args.repo)
+
+    if args.pull_request:
+        pr = gh_repo.get_pull(args.pull_request)
+
+        reviews = pr.get_reviews()
+        print(f'#{pr.number}: {pr.title} - {pr.comments} Comments, reviews: {reviews.totalCount}, {len(pr.assignees)} Assignees (Updated {pr.updated_at})')
+        assignee_reviews = 0
+        reviewers = set()
+        for r in reviews:
+            if r.user and r.state == 'APPROVED':
+                reviewers.add(r.user.login)
+            if pr.assignee and r.user:
+                if r.user.login == pr.assignee.login:
+                    assignee_reviews = assignee_reviews + 1
+        prj = {}
+        prj['nr'] = pr.number
+        prj['url'] = pr.url
+        prj['title'] = pr.title
+        prj['comments'] = pr.comments
+        prj['reviews'] = reviews.totalCount
+        prj['assignees'] = len(pr.assignees)
+        prj['updated'] = pr.updated_at.strftime("%Y-%m-%d %H:%M:%S")
+        prj['created'] = pr.created_at.strftime("%Y-%m-%d %H:%M:%S")
+        prj['closed'] = pr.closed_at.strftime("%Y-%m-%d %H:%M:%S")
+        prj['merged_by'] = pr.merged_by.login
+        prj['submitted_by'] = pr.user.login
+        prj['changed_file'] = pr.changed_files
+        prj['additions'] = pr.additions
+        prj['deletions'] = pr.deletions
+        prj['commits'] = pr.commits
+        prj['base'] = pr.base.ref
+
+        if pr.assignee:
+            prj['assignee'] = pr.assignee.login
+            prj['reviewed_by_assignee'] = "yes" if pr.assignee.login in reviewers else "no"
+            if pr.assignee.login in reviewers or 'Hotfix' in pr.labels or pr.assignee.login == pr.user.login:
+                prj['review_rule'] = "yes"
+            else:
+                prj['review_rule'] = "no"
+        else:
+            prj['assignee'] = "none"
+            prj['reviewed_by_assignee'] = "na"
+            prj['review_rule'] = "na"
+        prj['assignee_reviews'] = assignee_reviews
+        ll = []
+        for l in pr.labels:
+            ll.append(l.name)
+
+        prj['labels'] = ll
+        delta = pr.closed_at - pr.created_at
+        deltah = delta.total_seconds() / 3600
+        prj['hours_open'] = deltah
+
+        if deltah < 48 and not ('Trivial' in ll or 'Hotfix' in ll):
```

Comment:
Even if it won't/can't be perfect, I think this nice to include some notion of "business days" in the computation of the "minimum of 2 days" -- like maybe just ignore Saturday and Sunday "GMT"? 

There is certainly more than a handful of PRs merged on Mondays and early (european) Tuesdays that have either been open during the weekend or late on the previous Friday, and reporting them as being OK is not helping surface this issue.

Reply:
addressed.

Reply:
sorry if I should be able to answer my own question by just looking at and understanding the code but, this is now implemented as "2 fully elapsed business days", is it (which I'm fine with fwiw, it's maybe not just necessarily how people interpret the rule today) ? I.e a PR opened on a Friday GMT should not be merged until the following Wednesday, GMT, no. matter the actual time of the day it was opened/merged?

Reply:
huh, so if I open a PR on monday noon, what does 2 business days mean? The way I see it,  Wednsday  noon. If something was opened Friday at 15:00, mergeable it will be on Tuesday 15:00. That is what the script does.

```
from datetime import date, timedelta, datetime

start = datetime(2024, 1, 5, 15, 32, 0)
end = datetime(2024, 1, 9, 15, 32, 0)

all_days = (start + timedelta(x + 1) for x in range((end - start).days))

count = sum(1 for day in all_days if day.weekday() < 5)
print('Number of business days is:', count)
```

Reply:
> maybe not necessarily how people interpret the rule today?

based on my observation, most of us today look at what github says, ie.. 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/180017/0a8efbfd-393a-4b3c-8e0e-4235300d59c5)

which is never accurate and does take business days into account, so the above in terms of tracking is an improvement, in terms of knowing when exactly the "safe zone" is going to be tricky and will not be accurate (give or take few hours) which is fine. 
We do not want to have mergers set with a calculator and calendars calculating when they can merge something, this will be counter productive in every way possible.

Reply:
> That is what the script does.

That's perfect then! The `(end - start).days` bit was confusing me and making me think the time info was being lost.

And to clarify, this is absolutely how I *would* expect people to understand the rule indeed ; my comment was made as I wasn't sure whether that's what the current implementation was doing and if really was looking at "48 hours elapsed during business days".

Thanks!